python/private/attributes.bzl

```diff
@@ -267,11 +222,10 @@ COMMON_ATTRS = union_attrs(
         # buildifier: disable=attr-license
         "licenses": attr.license() if hasattr(attr, "license") else attr.string_list(),
     },
-    allow_none = True,
 )
 
 IMPORTS_ATTRS = {
-    "imports": attr.string_list(
+    "imports": lambda: attrb.StringList(
```

Comment:
hmmm... the `key: lambda: attrb.StringList(` does look funny :D

Reply:
Yeah, I don't see a way around it, though.

Unless, attrb.StringList() returned a lambda? ehhhh

```
ATTRS = {
    "imports": attrb.StringList(...)
}
```

That `ATTRS["imports"]` is a lambda is a bit surprising. The "imperative" case then becomes a bit convoluted:

```
  srcs = attrb.StringList()
  srcs = srcs()
  srcs.set_whatever()
```

It'd be simple enough to have two structs, one that returns a lambda, and another that didn't? e.g. `attrcb.StringList()`

I feel like the explicit lambdas are better overall. They help make explicit that there's a callback.

Reply:
Explicit is better than implicit, definitely.